libraries/AP_Scripting/lua_scripts.cpp

```diff
@@ -223,12 +231,14 @@ void lua_scripts::run_next_script(lua_State *L) {
                        gcs().send_text(MAV_SEVERITY_CRITICAL, "Lua: %s did not return a delay (0x%d)", script->name, lua_type(L, -1));
                        lua_pop(L, 2);
                        remove_script(L, script);
+                       reload = true; // Lua script bug, reload
                        return;
                    }
                    if (lua_type(L, -2) != LUA_TFUNCTION) {
                        gcs().send_text(MAV_SEVERITY_CRITICAL, "Lua: %s did not return a function (0x%d)", script->name, lua_type(L, -2));
                        lua_pop(L, 2);
                        remove_script(L, script);
+                       reload = true; // Lua script bug, reload
```

Comment:
I guess in this case we might actual have ended the script on purpose, probably that is not good scripting practice tho, you end up with the no scripts to run spaming.

Reply:
Ending the script on purpose is returning 0 args, which is explicitly handled above. I think that's fine.